django/db/backends/sqlite3/base.py

```diff
@@ -452,6 +449,7 @@ def _sqlite_time_extract(lookup_type, dt):
     return getattr(dt, lookup_type)
 
 
+@none_guard
```

Comment:
Removing this won't cause any test failures because the `except` clause below will catch the `TypeError` if there is `None` in the addition or subtraction. This will perform the check for `None` up front, however, avoiding executing all the calls to `typecast_timestamp()`.